datetimeparser/evaluatorutils.py

```diff
@@ -67,23 +67,23 @@ def sanitize_input(current_time: datetime, parsed_list: list) -> list[Union[Rela
         """
 
         ps = parsed_list
-        for index, element in enumerate(parsed_list):
+        for i, element in enumerate(parsed_list):
             if isinstance(element, Constant) and element.name == "of":
-                if isinstance(ps[index - 1], RelativeDateTime):
-                    if ps[index - 1].years != 0:
-                        ps[index - 1].years -= 1
-                    if ps[index - 1].months != 0:
-                        ps[index - 1].months -= 1
-                    if ps[index - 1].weeks != 0:
-                        if ps[index + 1] in MonthConstants.ALL:
+                if isinstance(ps[i - 1], RelativeDateTime):
+                    if ps[i - 1].years != 0:
+                        ps[i - 1].years -= 1
+                    if ps[i - 1].months != 0:
+                        ps[i - 1].months -= 1
+                    if ps[i - 1].weeks != 0:
+                        if ps[i + 1] in MonthConstants.ALL:
```

Comment:
Also, make use of `and`. I disabled the option in the linter but that would represent a (in this case relative not of the whole function) complexity of 2 (meaning you go 2 indentations deeper).
You should avoid such complexity most of the time. If it's over our 140 character limit, thats fine